packages/atomic/src/components/search/atomic-refine-modal/atomic-refine-modal.tsx

```diff
@@ -126,9 +132,115 @@ export class AtomicRefineModal implements InitializableComponent {
     this.sort = buildSort(this.bindings.engine);
     this.querySummary = buildQuerySummary(this.bindings.engine);
     this.searchStatus = buildSearchStatus(this.bindings.engine);
+    this.facetManager = buildFacetManager(this.bindings.engine);
     this.watchEnabled(this.isOpen);
   }
 
+  private createFacetSlot(): HTMLDivElement {
+    const divSlot = document.createElement('div');
+    divSlot.setAttribute('slot', 'facets');
+    this.addElementStyle(divSlot);
+
+    const facets = this.queryCurrentFacets();
+
+    const sortedFacets = sortFacetsViaManager(facets, this.facetManager);
+
+    const {visibleFacets, invisibleFacets} = sortFacetVisibility(
+      sortedFacets,
+      this.bindings.store.getAllFacets()
+    );
+
+    const visibleFacetsClone = this.cloneFacets(visibleFacets);
+    const invisibleFacetsClone = this.cloneFacets(invisibleFacets);
+
+    updateCollapsedState(visibleFacetsClone, this.collapseFacetsAfter);
+
+    divSlot.append(...visibleFacetsClone);
+    divSlot.append(...invisibleFacetsClone);
+
+    const generator = this.makeAutomaticFacetGenerator();
+    if (generator) {
+      updateCollapseFacetsAfter(
+        generator,
+        visibleFacets.length,
+        this.collapseFacetsAfter
+      );
+      divSlot.append(generator);
+    }
+
+    return divSlot;
+  }
+
+  private cloneFacets(facets: BaseFacetElement[]): BaseFacetElement[] {
+    return facets.map((facet) => {
+      facet.classList.remove(popoverClass);
+      facet.setAttribute(isRefineModalFacet, '');
+      const clone = facet.cloneNode(true) as BaseFacetElement;
+      clone.isCollapsed = true;
+      return clone;
+    });
+  }
+
+  private makeAutomaticFacetGenerator() {
+    if (!this.bindings.engine.state.automaticFacetSet?.desiredCount) {
+      return;
+    }
+    const generator = document.createElement(
+      'atomic-automatic-facet-generator'
+    );
+
+    generator.setAttribute(
+      'desired-count',
+      `${this.bindings.engine.state.automaticFacetSet?.desiredCount}`
+    );
+    this.addElementStyle(generator);
+
+    return generator;
+  }
+
+  private queryCurrentFacets(): BaseFacetElement[] {
+    const facetIds = Object.keys(this.bindings.store.getAllFacets());
+    const facetElements: BaseFacetElement[] = [];
+
+    facetIds.forEach((facetIdValue) => {
+      const matchingElement = document.querySelector(
```

Comment:
**Warning**: we should never use `querySelector` from the root `document`. We should assume there might be multiple search interfaces in the page.

Reply:
Why are we using `this.bindings.store.getAllFacets()` and mapping them to elements instead of directly calling `this.bindings.store.getFacetElements()`?